cluster/gce/gci/configure-helper-testcase.go

```diff
@@ -0,0 +1,200 @@
+}
+
+func NewManifestTestCase(manifestUnderTest, funcUnderTest string, auxManifests []string, t *testing.T) *ManifestTestCase {
+	c := &ManifestTestCase{}
+	c.t = t
+	d, err := ioutil.TempDir("/tmp", "")
+	if err != nil {
+		c.t.Fatalf("Failed to create temp directory: %v", err)
+	}
+
+	c.kubeHome = d
+	glog.V(3).Infof("KUBE_HOME of the test case is %s", c.kubeHome)
+
+	c.manifestFuncName = funcUnderTest
+	glog.V(3).Infof("configure-helper.sh function under test is: %s", c.manifestFuncName)
+
+	c.manifest = manifestUnderTest
+	glog.V(3).Infof("Manifest under test is %s", manifestUnderTest)
+
+	// Some manifests consume other manifests. For example, kube-apiserver.manifest may consume kms-plugin-container.manifest.
+	c.auxManifests = auxManifests
+	glog.V(3).Infof("Auxiliary manifests: %v", c.auxManifests)
+
+	c.envScriptPath = filepath.Join(c.kubeHome, envScriptFileName)
+	glog.V(3).Infof("kube-env variables will be sourced by bash from %s", c.envScriptPath)
+
+	c.manifestSources = path.Join(filepath.Join(c.kubeHome, "kube-manifests", "kubernetes", "gci-trusty"))
+	glog.V(3).Infof("Function under test will access the manifest from %s", c.manifestSources)
+
+	// full path of this go file
+	_, testFilePath, _, _ := goruntime.Caller(0)
+	c.manifestTemplateDir = filepath.Join(path.Dir(path.Dir(testFilePath)), "manifests")
+	c.manifestTemplate = filepath.Join(c.manifestTemplateDir, c.manifest)
+	glog.V(3).Infof("Test case will use a copy of this manifest: %s", c.manifestTemplate)
+
+	c.manifestDestination = filepath.Join(c.kubeHome, "etc", "kubernetes", "manifests", c.manifest)
+	glog.V(3).Infof("The resulting manifest will be created in %s", c.manifestDestination)
+
+	c.mustCopyFromTemplate()
+	c.mustCopyAuxFromTemplate()
+	c.mustCreateManifestDstDir()
+
+	return c
+}
+
+func (c *ManifestTestCase) mustCopyFromTemplate() {
+	err := os.MkdirAll(c.manifestSources, os.ModePerm)
+	if err != nil {
+		c.t.Fatalf("Failed to create source directory: %v", err)
+	}
+
+	err = copyFile(c.manifestTemplate, filepath.Join(c.manifestSources, c.manifest))
+
+	if err != nil {
+		c.t.Fatalf("Failed to copy source manifest to KUBE_HOME: %v", err)
+	}
+}
+
+func (c *ManifestTestCase) mustCopyAuxFromTemplate() {
+	for _, m := range c.auxManifests {
+		err := copyFile(filepath.Join(c.manifestTemplateDir, m), filepath.Join(c.manifestSources, m))
+		if err != nil {
+			c.t.Fatalf("Failed to copy source manifest %s to KUBE_HOME: %v", m, err)
+		}
+	}
+}
+
+func (c *ManifestTestCase) mustCreateManifestDstDir() {
+	p := path.Join(filepath.Join(c.kubeHome, "etc", "kubernetes", "manifests"))
+	err := os.MkdirAll(p, os.ModePerm)
+	if err != nil {
+		c.t.Fatalf("Failed to create designation folder for kube-apiserver.manifest: %v", err)
+	}
+}
+
+func (c *ManifestTestCase) mustCreateEnv(envTemplate string, env interface{}) {
+	f, err := os.Create(filepath.Join(c.kubeHome, envScriptFileName))
+	if err != nil {
+		c.t.Fatalf("Failed to create envScript: %v", err)
+	}
```

Comment:
defer f.Close() after checking the error

Reply:
Done.